linkcheck/parser/__init__.py

```diff
@@ -100,8 +100,11 @@ def parse_css (url_data):
 def parse_swf (url_data):
     """Parse a SWF file for URLs."""
     linkfinder = linkparse.swf_url_re.finditer
-    for mo in linkfinder(url_data.get_content()):
-        url = mo.group()
+    for mo in linkfinder(url_data.get_raw_content()):
+        # We're scraping binary data for anything that looks like an URL using
+        # a regex that matches only ASCII characters.  Any non-ASCII characters
+        # in the URL are expected to be %-encoded.
+        url = mo.group().decode('ascii')
```

Comment:
I think it might even work without this because of:
https://github.com/linkchecker/linkchecker/blob/125146fb2c11956cf492d2db82d2d3644a4ad520/linkcheck/checker/__init__.py#L98
via:
https://github.com/linkchecker/linkchecker/blob/125146fb2c11956cf492d2db82d2d3644a4ad520/linkcheck/checker/urlbase.py#L685-L691

But, I prefer decoding data at the edge as per this change, maybe in future it means we can get rid of the more internal decoding